library/lib/application.cpp

```diff
@@ -229,23 +229,126 @@ bool Application::init(std::string title, Style* style, LibraryViewsThemeVariant
 #ifdef __SWITCH__
     {
         PlFontData font;
+        int locale = i18n::swGetCurrentLocaleID();
 
         // Standard font
         Result rc = plGetSharedFontByType(&font, PlSharedFontType_Standard);
         if (R_SUCCEEDED(rc))
         {
-            Logger::info("Using Switch shared font");
-            Application::fontStash.regular = Application::loadFontFromMemory("regular", font.address, font.size, false);
+            Logger::info("Adding Switch shared standard font");
+            Application::fontStash.standard = Application::loadFontFromMemory("regular", font.address, font.size, false);
         }
 
-        // Korean font
-        rc = plGetSharedFontByType(&font, PlSharedFontType_KO);
-        if (R_SUCCEEDED(rc))
+        // Load other fonts on demand
+        bool isFullFallback = false;
+        AppletType at = appletGetAppletType();
+        if (at == AppletType_Application || at == AppletType_SystemApplication) // title takeover
+        {
+            isFullFallback = true;
+            Logger::warning("Non applet mode, font full fallback is enabled!");
+        }
+        
+        if (locale == 6 || locale == 15 || isFullFallback)
```

Comment:
Not a fan of magic values here, could you define them somewhere instead?

Reply:
substituted with enforcing `SetLanguage` enum